components/service/glean/src/main/java/mozilla/components/service/glean/CommonMetricData.kt

```diff
@@ -74,4 +74,18 @@ interface CommonMetricData {
         val filteredNames = sendInPings.filter { it != DEFAULT_STORAGE_NAME }
         return filteredNames + defaultStorageDestinations
     }
+
+    /**
+     * Returns a copy of this metric with the name changed to `newName`.
+     *
+     * This is used by |LabeledMetricType| to create metrics that belong
+     * in labeled metric groups.  Must be overridden by all metric types
+     * that can be labeled.
+     *
+     * @param newName The new name
+     * @returns A copy of this metric with the name changed to `newName`.
+     */
+    fun getWithName(newName: String): T {
```

Comment:
I wish it were possible to make this `internal`, but apparently it isn't. As a result, this is showing up in the metric API. Do you have any idea on how to fix that? I would also call this `getWithNewName` :) But that's bikeshedding, up to you

Reply:
Thanks for the suggestion, @travis79.  I ended up solving this in a different way with a big delegation switch function (which can easily be private so encapsulates things the way we want).